crates/core/src/polynomial/arith_circuit.rs

```diff
@@ -123,7 +124,7 @@ enum CircuitStep<F: Field> {
 /// and the object representing different polnomials can be stored in a homogeneous collection.
 #[derive(Debug, Clone)]
 pub struct ArithCircuitPoly<F: Field> {
-	expr: ArithExpr<F>,
+	pub(crate) expr: ArithExpr<F>,
```

Comment:
Instead of making this field public, I recommend adding a method `pub fn expr(&self) -> &ArithExpr<F>` since that makes it easier to reason about potential mutation of the field. You can do this slightly less verbosely using the `Getters` derive macro.